### ISSUE TITLE
(WCAG 2.1) Lack of Screen Reader Announcement for Error Message on form submission

### ISSUE THREAD

**Issue by dkatzz** [civiform/civiform#5519]
OBSERVED RESULTS:

- When the "Save and Next" button is activated with empty input fields or erroneous data, an error message is displayed on the screen. However, the availability of the error message is not being announced to screen reader users.
- Additionally, the focus does not move to the first field with an error upon activating the "Save and Next" button.
- Applicable for all the forms in the “Comprehensive sample program" application steps.

EXPECTED RESULTS:

- Upon activating the "Save and Next" button with empty or erroneous data, the error message should be present on the top of page and announced to screen reader users.
- The focus should automatically move to the first field with an error so that screen reader users can easily identify and correct the problem.
- Implement an aria-live region or similar accessibility technique to ensure that screen reader users are informed about the presence of an error message when they activate the "Save and Next" button.
- After displaying the error message, programmatically move the focus to the first input field with an error using JavaScript.

STEPS TO REPRODUCE:

- Enable screen reader.
- Navigate to [URL](https://demo.civiform.dev/)
- Navigate to and activate the ‘Apply’ for the Comprehensive sample program.
- Navigate to and activate “What is your address?”.
- Without filling any of the input fields, Navigate to and activate “Save and Next” .
- Observe that the error message or presence of error message is not announced to the screen reader user.
- The focus does not move to the first input field with an error.

ENVIRONMENT DETAILS:

- Device(s) (brand, OS version, test key or release key): Windows 10 Enterprise (64-bit),MacOS Ventura Version 13.4
- Screen reader name and version: JAWS 2022.2211.7, NVDA 2022.3, VoiceOver
- Browser name and version: Chrome Version 113.0.5672.93

![Screenshot 1](https://github.com/civiform/civiform/assets/86739416/6da83a9b-9c5f-49b5-b504-07e591e5d38f)
![Screenshot 2](https://github.com/civiform/civiform/assets/86739416/c1cc1168-1aaa-4fa7-b542-163b5ccca183)
![Screenshot 3](https://github.com/civiform/civiform/assets/86739416/673194ca-052d-4482-aa25-151a5db061d9)
![Screenshot 4](https://github.com/civiform/civiform/assets/86739416/edf7b14f-d9d2-4be7-8ce0-d90ecd43f29b)
![Screenshot 5](https://github.com/civiform/civiform/assets/86739416/d847abf2-e7ef-4ab9-89cc-58291bc945a4)
